pilot/pkg/model/service.go

```diff
@@ -193,6 +193,10 @@ type Port struct {
 
 	// Protocol to be used for the port.
 	Protocol protocol.Instance `json:"protocol,omitempty"`
+
+	// The port number on the endpoint where the traffic will be received.
+	// It can map the k8s service or ServiceEntries with targetPort specified
+	TargetPort int `json:"target_port,omitempty"`
```

Comment:
This is not how the internal model is intended to be used. TargetPort only makes sense in the context of a ServiceInstance which already has this (EndpointPort)

A service does not just have one singular int target port

Reply:
If so, target port will be missing when using `ServiceToServiceEntry` [here ](https://github.com/istio/istio/blob/b55bf1a7debc0ce0081e91ad25694e839737512b/pilot/pkg/networking/apigen/apigen.go#L129). 
One question, could I pass another parameter with ServiceInstance  for `ServiceToServiceEntry` to fetch the target port?

Reply:
I think there are multiple levels of problems with that function

1. SE -> service -> is lossy
2. ServiceEntry cannot fully describe a Kubernetes Service (which allows target port by name)

Reply:
1. SE -> service -> is lossy  
Yes
2. ServiceEntry cannot fully describe a Kubernetes Service (which allows target port by name) 
Yes, that's why there are so many `TODO`s in this function, I think istio service and proxy can not really convert corresponding ServiceEntry.

Reply:
BTW, I think all necessary information should be passed to `ServiceToServiceEntry` to complete this conversion, what's your suggestion? @howardjohn

Reply:
I suggest we donot add this unless we need it internally, in case of abuse later.

Reply:
Okay, @hzxuzhonghu, I will try another way to get the target port.

Reply:
> al model is intended to be used. TargetPort only makes sense in the context of a ServiceInstance which already has this (EndpointPort)

How about we make TargetPort can be either a number or a name? Just like k8s svc.

Reply:
> This is not how the internal model is intended to be used. TargetPort only makes sense in the context of a ServiceInstance which already has this (EndpointPort)
> 
> A service does not just have one singular int target port

@howardjohn How about we make TargetPort can be either a number or a name? Just like k8s svc.